src/databricks/sql/utils.py

```diff
@@ -96,11 +102,33 @@ def build_queue(
         else:
             raise AssertionError("Row set type is not valid")
 
+class ColumnQueue(ResultSetQueue):
+    def __init__(self, columnar_table, column_names):
+        self.columnar_table = columnar_table
+        self.cur_row_index = 0
+        self.n_valid_rows = len(columnar_table[0])
+        self.column_names = column_names
+
+    def next_n_rows(self, num_rows):
+        length = min(num_rows, self.n_valid_rows - self.cur_row_index)
+        # Slicing using the default python slice
+        next_data = [
+            column[self.cur_row_index : self.cur_row_index + length]
+            for column in self.columnar_table
+        ]
+        self.cur_row_index += length
+        return next_data
+
+    def remaining_rows(self):
```

Comment:
Is this standard naming? The name seems this is read only operation

Reply:
The names are defined in the already existing ResultSetQueue from which I inherited